compiler/resolution/functionResolution.cpp

```diff
@@ -4145,7 +4165,20 @@ static void generateUnresolvedMsg(CallInfo& info, Vec<FnSymbol*>& visibleFns) {
     USR_FATAL_CONT(call,
                    "unresolved enumerated type symbol or call '%s'",
                    str);
-
+  } else if (isInitEqualsPreResolve(call)) {
+    INT_ASSERT(info.actuals.v[0]->getValType() == dtMethodToken);
+
+    // TODO: Not sure if this is enforced, should we remove?
+    INT_ASSERT(info.actuals.n == 3);
```

Comment:
I think this is at least partially enforced with https://github.com/chapel-lang/chapel/blob/master/compiler/AST/AggregateType.cpp#L599-L604 (assuming that we only add `this` and the method token afterwards)

However, I'd also keep this assertion just in case. And in that case, I'd remove the comment above.

Reply:
Thanks!